osutil/disks/disks_test.go

```diff
@@ -185,12 +249,68 @@ fi
 	d, err := disks.DiskFromMountPoint("/run/mnt/point", nil)
 	c.Assert(err, IsNil)
 	c.Assert(d.Dev(), Equals, "42:1")
+	c.Assert(d.HasPartitions(), Equals, false)
 
 	c.Assert(udevadmCmd.Calls(), DeepEquals, [][]string{
 		{"udevadm", "info", "--query", "property", "--name", "/dev/mapper/something"},
 	})
 }
 
+func (s *diskSuite) TestDiskFromMountPointIsDecryptedDeviceVolumeHappy(c *C) {
+	restore := osutil.MockMountInfo(`130 30 242:1 / /run/mnt/point rw,relatime shared:54 - ext4 /dev/mapper/something rw
+`)
+	defer restore()
+
+	restore = disks.MockUdevPropertiesForDevice(func(dev string) (map[string]string, error) {
+		switch dev {
+		case "/dev/mapper/something":
+			return map[string]string{
+				"DEVTYPE": "disk",
+			}, nil
+		case "/dev/disk/by-uuid/5a522809-c87e-4dfa-81a8-8dc5667d1304":
+			return map[string]string{
+				"DEVTYPE": "disk",
+			}, nil
+		default:
+			c.Logf("unexpected udev device properties requested: %s", dev)
+			c.Fail()
+			return nil, fmt.Errorf("unexpected udev device")
+
+		}
+	})
+	defer restore()
+
+	// mock the /sys/dev/block dir
+	devBlockDir := filepath.Join(dirs.SysfsDir, "dev", "block")
+	restore = disks.MockDevBlockDir(devBlockDir)
+	defer restore()
+
+	// mock the sysfs dm uuid and name files
+	dmDir := filepath.Join(devBlockDir, "242:1", "dm")
+	err := os.MkdirAll(dmDir, 0755)
+	c.Assert(err, IsNil)
+
+	err = ioutil.WriteFile(
+		filepath.Join(dmDir, "name"),
+		[]byte("something"),
+		0644,
+	)
+	c.Assert(err, IsNil)
+
+	err = ioutil.WriteFile(
```

Comment:
(nitpick^2) Hm, slightly unusual wrapping (here and in two more places below), maybe just:
```
	s := "CRYPT-LUKS2-5a522809c87e4dfa81a88dc5667d1304-something"
	err = ioutil.WriteFile(filepath.Join(dmDir, "uuid"), []byte(s), 0644)
```